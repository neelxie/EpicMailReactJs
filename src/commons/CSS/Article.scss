
.article-box {
  border-radius: 0;
  border-width: 0;
  font-size: 14px;
  width: 350px;

  img {
    border-radius: 0;
    height: 183px;
  }

  .card-body {
    padding: 1.25rem 0;
  }

  .card-title {
    color: $default-gray;
    font-size: 14px;
    font-style: normal;
    font-weight: bold;
    margin-bottom: .1rem;
    text-align: left;
  }

  .card-text,
  .card-link {
    color: $default-gray;
    font-family: $lato-font;
    font-weight: 100;
    margin-bottom: .2rem;
    max-width: 43ch;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  .row {
    margin-top: -5px;

    .left {
      text-align: left;
    }

    .right {
      text-align: right;
    }
  }

  .dv-star-rating-star {
    font-size: 1.2em !important;
    margin-left: -3px;
    padding: 2px 2px 0 2px;
  }

  .dv-star-rating-full-star {
    color: #121111 !important;
  }

  .dv-star-rating-empty-star {
    color: $light-gray !important;
  }

  label {
    margin-bottom: 0;
  }
}
