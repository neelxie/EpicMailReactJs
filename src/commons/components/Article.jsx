/* eslint-disable react/forbid-prop-types */
/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-unused-vars */
import React from 'react';
import '../CSS/Article.scss';
import { Link } from 'react-router-dom';
import { Card, Row, Col } from 'react-bootstrap';
import StarRatingComponent from 'react-star-rating-component';
import PropTypes from 'prop-types';

function Article({ article, article: { author } }) {
  const authors = article.author;
  let authorsName = 'author username';
  let createdAt = '';

  if (authors !== undefined) {
    authorsName = authors.username;
    createdAt = new Date(article.createdAt).toLocaleDateString('en-GB', {
      day: 'numeric',
      month: 'short',
      year: 'numeric',
    });
  }

  return (
    <div>
      <Card className="article-box">
        <Card.Img variant="top" src={article.image} />
        <Card.Body>
          <Link to={`/article/${article.slug}`}>
            <Card.Title>
              {article.title}
            </Card.Title>
          </Link>
          <Card.Text>
            {article.body}
          </Card.Text>
          <StarRatingComponent
            name="rate2"
            editing={false}
            renderStarIcon={() => <span>â˜…</span>}
            starCount={5}
            value={article.average_rating}
          />
          <Row>
            <Col xs={6} className="left">
              <Card.Link href="#">
                {authorsName}
              </Card.Link>
            </Col>
            <Col xs={6} className="right">
              <Card.Link href="#">
                {createdAt}
              </Card.Link>
            </Col>
          </Row>
        </Card.Body>
      </Card>
    </div>
  );
}

Article.propTypes = {
  article: PropTypes.object,
};


Article.defaultProps = {
  article: {},
};

export default Article;
