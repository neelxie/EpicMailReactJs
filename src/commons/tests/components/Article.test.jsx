/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Article from '../../components/Article';

function renderArticleComponent(args) {
  const defaultProps = {
    article: {},
  };

  const props = { ...defaultProps, ...args };
  return shallow(<Article article={props} />);
}

describe('Article Component', () => {
  it('should match the snapshot', () => {
    const component = renderArticleComponent();

    expect(toJson(component)).toMatchSnapshot();
  });

  it('article should have the title passed to it  ', () => {
    const component = renderArticleComponent({
      title: 'i am who I am',
      image: 'http://url.img',
      createsAt: '',
      author: {
        username: 'alexxsanya',
      },
    });
    expect(component.find('CardTitle').text()).toEqual('i am who I am');
  });
});
