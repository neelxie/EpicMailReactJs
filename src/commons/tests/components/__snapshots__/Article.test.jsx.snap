// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Component should match the snapshot 1`] = `
<div>
  <Card
    body={false}
    className="article-box"
  >
    <CardImg
      variant="top"
    />
    <CardBody>
      <Link
        to="/article/undefined"
      >
        <CardTitle />
      </Link>
      <CardText />
      <StarRatingComponent
        editing={false}
        emptyStarColor="#333"
        name="rate2"
        renderStarIcon={[Function]}
        starColor="#ffb400"
        starCount={5}
      />
      <ForwardRef
        noGutters={false}
      >
        <Col
          className="left"
          xs={6}
        >
          <CardLink
            href="#"
          >
            author username
          </CardLink>
        </Col>
        <Col
          className="right"
          xs={6}
        >
          <CardLink
            href="#"
          />
        </Col>
      </ForwardRef>
    </CardBody>
  </Card>
</div>
`;
